lib/rouge/lexers/pony.rb

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Pony < RegexLexer
+      tag 'pony'
+      filenames '*.pony'
+
+      keywords = %w(
+        actor addressof and as
+        be break
+        class compiler_intrinsic consume continue
+        do
+        else elseif embed end error
+        for fun
+        if ifdef in interface is isnt
+        lambda let
+        match
+        new not
+        object
+        primitive
+        recover repeat return
+        struct
+        then this trait try type
+        until use
+        var
+        where while with
+      )
+
+      capabilities = %w(
+        box iso ref tag trn val
+      )
+
+      types = %w(
+        Number Signed Unsigned Float
+        I8 I16 I32 I64 I128 U8 U32 U64 U128 F32 F64
+        EventID Align IntFormat NumberPrefix FloatFormat
+        Type
+      )
+
+      state :whitespace do
+        rule /[\s\t\r\n]+/m, Text
+      end
+
+      state :root do
+        mixin :whitespace
+        rule /"""/, Str::Doc, :docstring
+        rule /"/, Str, :string
+        rule %r(//(.*?)\n), Comment::Single
+        rule %r(/(\\\n)?[*](.|\n)*?[*](\\\n)?/), Comment::Multiline
+        rule /#{capabilities.join('|')}\b/, Keyword::Declaration
```

Comment:
Rather than creating massive chained regexes here, it's preferable to match a single identifier regex and check for inclusion in various sets. As an example, the javascript lexer does this here: https://github.com/jneen/rouge/blob/master/lib/rouge/lexers/javascript.rb#L178